specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-11-01-preview/elastic.json

```diff
@@ -0,0 +1,2793 @@
+          "maxLength": 50,
+          "type": "string"
+        },
+        "lastName": {
+          "description": "Last name of the user",
+          "maxLength": 50,
+          "type": "string"
+        },
+        "companyName": {
+          "description": "Company name of the user",
+          "maxLength": 64,
+          "type": "string"
+        },
+        "emailAddress": {
+          "description": "Email of the user used by Elastic for contacting them if needed",
+          "pattern": "^([^<>()\\[\\]\\.,;:\\s@\"]+(\\.[^<>()\\[\\]\\.,;:\\s@\"]+)*)@(([a-zA-Z-_0-9]+\\.)+[a-zA-Z]{2,})$",
+          "type": "string"
+        },
+        "companyInfo": {
+          "description": "Company information of the user to be passed to partners.",
+          "$ref": "#/definitions/CompanyInfo"
+        }
+      }
+    },
+    "PlanDetails": {
+      "description": "Plan details of the monitor resource.",
+      "type": "object",
+      "properties": {
+        "offerID": {
+          "description": "Offer ID of the plan",
+          "type": "string"
+        },
+        "publisherID": {
+          "description": "Publisher ID of the plan",
+          "type": "string"
+        },
+        "termID": {
+          "description": "Term ID of the plan",
+          "type": "string"
+        },
+        "planID": {
+          "description": "Plan ID",
+          "type": "string"
+        },
+        "planName": {
+          "description": "Plan Name",
+          "type": "string"
+        }
+      }
+    },
+    "LiftrResourceCategories": {
+      "enum": [
+        "Unknown",
+        "MonitorLogs"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "LiftrResourceCategories"
+      },
+      "readOnly": true
+    },
+    "MonitorProperties": {
+      "description": "Properties specific to the monitor resource.",
+      "type": "object",
+      "properties": {
+        "provisioningState": {
+          "description": "Provisioning state of the monitor resource.",
+          "$ref": "#/definitions/ProvisioningState"
+        },
+        "monitoringStatus": {
+          "description": "Flag specifying if the resource monitoring is enabled or disabled.",
+          "$ref": "#/definitions/MonitoringStatus"
+        },
+        "elasticProperties": {
+          "description": "Elastic cloud properties.",
+          "$ref": "#/definitions/ElasticProperties"
+        },
+        "userInfo": {
+          "description": "User information.",
+          "$ref": "#/definitions/UserInfo",
+          "x-ms-mutability": [
+            "create"
+          ]
+        },
+        "planDetails": {
+          "description": "Plan details of the monitor resource.",
+          "$ref": "#/definitions/PlanDetails"
+        },
+        "version": {
+          "description": "Version of elastic of the monitor resource",
+          "type": "string"
+        },
+        "saaSAzureSubscriptionStatus": {
```

Comment:
>saaSAzureSubscriptionStatus [](http://example.com/codeflow?start=9&length=27)

Should this be an enum?

Reply:
It doesn't have pre-defined value set, and this is an updatable field. Therefore, went with string.